### PR TITLE
Refactor score tests to make them less verbose and more direct

### PR DESCRIPTION
**Pull request by mattbowen-usds**

### CODE REVIEW

File: data/data-pipeline/data_pipeline/score/score_narwhal.py [hunk 1]
```diff
@@ -14,20 +14,17 @@
...
-        self.LACK_OF_HIGH_SCHOOL_MINIMUM_THRESHOLD: float = 0.10
-
-        # We define a donut hole DAC as a tract that is entirely surrounded by
-        # DACs (score threshold = 1) and above median for low income, as a starting
-        # point. As we ground-truth, these thresholds might change.
-        self.LOW_INCOME_THRESHOLD_DONUT: float = 0.50
-        self.SCORE_THRESHOLD_DONUT: float = 1.00
-
-        super().__init__(df)
+    LOW_INCOME_THRESHOLD: float = 0.65
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/score/score_narwhal.py:17]: this is clever!

File: data/data-pipeline/data_pipeline/tests/test_score_calculation.py [hunk 2]
```diff
@@ -1,302 +1,152 @@
...
+class PercentileTestConfig:
+    percentile_column_name: str
+    threshold_column_name: str
+    threshold: float
+    percentile_column_need_suffix: bool = True
 
-@pytest.fixture
-def lead_paint_proxy_home_value_threshold():
-    return ScoreNarwhal(final_score_df).MEDIAN_HOUSE_VALUE_THRESHOLD
+    @property
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/tests/test_score_calculation.py:18]: This is muuuuch nicer and clearer.

File: data/data-pipeline/data_pipeline/tests/test_score_calculation.py [hunk 3]
```diff
@@ -1,302 +1,152 @@
...
-    final_score_df,
-    env_percentile_threshold,
-):
-    thresholded_column, pctile_column = expected_pop_loss_percentile_columns
-    _helper_test_percentile_against_threshold(
-        final_score_df,
-        pctile_column,
-        thresholded_column,
-        env_percentile_threshold,
+    errors = []
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/tests/test_score_calculation.py:44]: This is also really clever

File: data/data-pipeline/data_pipeline/tests/test_score_calculation.py [hunk 4]
```diff
@@ -1,302 +1,152 @@
...
+        return self.percentile_column_name
 
 
 ### TODO: we need to blow this out for all eight categories
-def _helper_test_percentile_against_threshold(
-    df, pctile_column, thresholded_column, threshold
-):
+def _check_percentile_against_threshold(df, config: PercentileTestConfig):
     """Note - for the purpose of testing, this fills with False"""
     minimum_flagged = (
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/tests/test_score_calculation.py:31]: I did a poor job naming these variables. We might want them to be `minimum_flag`ok` or something to indicate that they are booleans?

File: data/data-pipeline/data_pipeline/tests/test_score_calculation.py [hunk 5]
```diff
@@ -1,302 +1,152 @@
...
+            f"For column {config.threshold_column_name}, there is someone flagged below {config.threshold} percentile!"
+        )
+    if not maximum_not_flagged:
+        errors.append(
+            f"For column {config.threshold_column_name}, there is someone not flagged above {config.threshold} percentile!"
+        )
+    return errors
+
+
+def test_percentile_columns(final_score_df):
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/tests/test_score_calculation.py:56]: this is extremely clean

File: data/data-pipeline/data_pipeline/tests/test_score_calculation.py [hunk 6]
```diff
@@ -1,302 +1,152 @@
...
+    donut_hole_income = PercentileTestConfig(
+        field_names.POVERTY_LESS_THAN_200_FPL_IMPUTED_FIELD,
+        field_names.FPL_200_SERIES_IMPUTED_AND_ADJUSTED_DONUTS,
+        ScoreNarwhal.LOW_INCOME_THRESHOLD_DONUT,
+    )
+    donut_hole_adjacency = PercentileTestConfig(
+        (field_names.SCORE_N_COMMUNITIES + field_names.ADJACENCY_INDEX_SUFFIX),
+        field_names.ADJACENT_TRACT_SCORE_ABOVE_DONUT_THRESHOLD,
+        ScoreNarwhal.SCORE_THRESHOLD_DONUT,
+        False,
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/tests/test_score_calculation.py:103]: can you add the `percentile_column_need_suffix` named arg here?

File: data/data-pipeline/data_pipeline/tests/test_score_calculation.py [hunk 7]
```diff
@@ -1,302 +1,152 @@
...
+        population_loss,
+        agricultural_loss,
+        building_loss,
+        flood,
+        wildfire,
+        low_high_school,
+        donut_hole_income,
+        donut_hole_adjacency,
+    ):
+        errors.extend(
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/tests/test_score_calculation.py:117]: this is sooooo nice

File: data/data-pipeline/data_pipeline/tests/test_score_output.py [hunk 8]
```diff
@@ -33,9 +23,7 @@ def climate_thresholds():
 
 @pytest.fixture
 def environmental_burden_threshold():
-    # well this is bad practice
-    narwhal_params = ScoreNarwhal(pd.DataFrame())
-    return narwhal_params.ENVIRONMENTAL_BURDEN_THRESHOLD
+    return ScoreNarwhal.ENVIRONMENTAL_BURDEN_THRESHOLD
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/tests/test_score_output.py:26]: yay! thank you for fixing this!

**Review by emma-nechamkin — COMMENTED** [review]
This looks sooo much better and I learned a lot looking through your code! Ty for implementing this in a much cleaner and better way -- I reaaaally appreciate it!

File: data/data-pipeline/data_pipeline/tests/test_score_calculation.py (hunk 2 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/tests/test_score_calculation.py:18]: I don't love that I have that `percentile_column_need_suffix` flag but also like, it's good enough for smoketests?

**Pull request merged**